### ISSUE TITLE
Implement additional types of filter predicates

### ISSUE THREAD

**Issue by vishakh** [Cryptonomic/Conseil#175]
The filters supported by the Conseil API currently represent a conjunction of simple predicates, i.e. "give me all `x` in set `A` AND all `y` in set `B`. We would not also like to support negations (all `x` not in set `A`) and comparators (`x > A`, `x = B` or `x < C). The API should therefore be extended to support these new types.

We might in the future also want to support 'OR' combinations, nested filters, etc so some thought should be given to building an extensible solution.

Depends on #183

**Comment by anonymoussprocket** [comment]
Superseded by #188
